core/storage/statistics/gae_models.py

```diff
@@ -1134,6 +1138,122 @@ def delete_playthroughs_multi(cls, playthrough_ids):
+            state_name: str. The name of the state.
+
+        Returns:
+            str. The state_reference for a new instance of this class.
+        """
+        return '%s.%s' % (exp_id, state_name)
+
+    @classmethod
+    def get_state_reference_for_question(cls, question_id):
+        """Generate the state_reference for the state in the question.
+
+        Args:
+            question_id: str. The id of the question.
+
+        Returns:
+            str. The state_reference for a new instance of this class.
+        """
+        return question_id
+
+    @classmethod
+    def create_model_instance(
+            cls, state_reference, entity_type, interaction_id,
+            learner_answer_info_list, schema_version,
+            accumulated_answer_info_json_size_bytes):
+        """Creates a new LearnerAnswerDetailsModel for the given entity type
+        then writes it to the datastore.
+
+        Args:
+            state_reference: str. The reference to the state for which model
+                instance is being created. For exploration state it will be of
+                the form 'exp_id.state_name', & for question it will be of the
+                form 'question_id'.
+            entity_type: str. The type of entity i.e "exploration" or
+                "question".
+            interaction_id: str.  The ID of the interaction for which the
+                answer details are received.
+            learner_answer_info_list: list. The list of LearnerAnswerInfo
+                objects in dict format, which is defined in the stats_domain.
+            schema_version: int. The version of LearnerAnswerInfo dict, which
+                is currently supported by the Oppia.
+            accumulated_answer_info_json_size_bytes: int. The size of the
+                learner_answer_info_list in bytes.
+        """
+        answer_details_instance = cls(
+            state_reference=state_reference,
+            entity_type=entity_type,
+            interaction_id=interaction_id,
+            learner_answer_info_list=learner_answer_info_list,
+            schema_version=schema_version,
+            accumulated_answer_info_json_size_bytes=(
+                accumulated_answer_info_json_size_bytes))
+        answer_details_instance.put()
+
+    @classmethod
+    def get_model_instance(
+            cls, entity_type, state_reference):
+        """Returns the model instance related to the entity type and
+        state reference.
+
+        Args:
+            entity_type: str. The type of the entity i.e "exploration" or
+                "question".
+            state_reference: str. The reference to a state, for which the model
+                is to be fetched. Foe exploration state it will be of the form
+                'exp_id.state_name', and for question state it will be of the
+                form 'question_id'.
+
+        Returns:
+            list(LearnerAnswerDetailsModel). List of answer details associated
```

Comment:
Since we are doing 1 model per state, it won't be a list anymore, right?

Reply:
Done